### PR TITLE
feat: implement target to dockerfile build

### LINKED ISSUE #4244: Implement target for Dockerfile build
This allows a user to build a particular target in the Dockerfile

```graphql
type Directory {
  dockerBuild(dockerfile: String, platform: Platform, target: String): Container!
}
```

### PR DESCRIPTION
**Pull request by grouville**
Closes #4244

Allow users to build a particular target in a multi-stage Dockerfile.

Extends these APIs:
- `(container *Container) Build`:
```go
output, err = c.Container().Build(src, dagger.ContainerBuildOpts{Target: "stage1"}).WithExec([]string{}).Stdout(ctx)
```
- `(s *containerSchema) build`:
```go
output, err = src.DockerBuild(dagger.DirectoryDockerBuildOpts{
	Target: "stage1",
}).WithExec([]string{}).Stdout(ctx)
```

This feature was recently requested by active users, on Discord: https://discord.com/channels/707636530424053791/708371226174685314/1061942068164964382

PS: the `.gitignore` diff is for me, as I'm relying on go workspaces to have the linter on VSCode. Did not want to make a single PR to add that.

Signed-off-by: grouville

### CODE REVIEW

**Review by jlongtine — APPROVED** [review]
LGTM!

File: core/schema/directory.graphqls [hunk 1]
```diff
@@ -64,10 +64,11 @@ type Directory {
     Path to the Dockerfile to use.
     Defaults to './Dockerfile'.
     """
-    dockerfile: String,
+    dockerfile: String
 
-    platform: Platform,
+    platform: Platform
     buildArgs: [BuildArg!]
+    target: String
```
Comment (vito) [inline core/schema/directory.graphqls:71]: Not sure why these are missing doc strings, here's a backfill:

```suggestion
    "The platform to build."
    platform: Platform
    "Additional build arguments."
    buildArgs: [BuildArg!]
    "Target build stage to build."
    target: String
```

**Review by vito — APPROVED** [review]
Looks perfect! Just noticed some missing doc strings from before this PR - I suggested changes but it'll probably need a SDK rebuild too. OK with merging this as-is either way.

File: core/schema/directory.graphqls (hunk 1 above)
Comment (grouville) [reply core/schema/directory.graphqls]: @vito, rebased with your suggested changes + a rebuild of the SDKs

**Pull request merged**